src/Enhavo/Bundle/SettingBundle/Entity/Setting.php

```diff
@@ -1,99 +1,220 @@
 <?php
 
 namespace Enhavo\Bundle\SettingBundle\Entity;
-
 use Doctrine\ORM\Mapping as ORM;
+use Sylius\Component\Resource\Model\ResourceInterface;
 
-/**
- * Setting
- */
-class Setting
+class Setting implements ResourceInterface
 {
+
+    const SETTING_TYPE_TEXT = 'text';
+    const SETTING_TYPE_BOOLEAN = 'boolean';
+    const SETTING_TYPE_FILE = 'file';
+    const SETTING_TYPE_FILES = 'files';
+    const SETTING_TYPE_WYSIWYG = 'wysiwyg';
+
+    protected $id;
+    protected $label;
+    protected $key;
+    protected $type;
+    protected $value;
+    protected $translationDomain;
+    protected $file;            // pointing to one file
+    protected $files;           // pointing to a collection of files
     /**
-     * @var integer
+     * Constructor
      */
-    protected $id;
+    public function __construct()
+    {
+        $this->files = new \Doctrine\Common\Collections\ArrayCollection();
+    }
 
     /**
-     * @var string
+     * Get id
+     *
+     * @return integer
      */
-    protected $name;
+    public function getId()
+    {
+        return $this->id;
+    }
 
     /**
-     * @var string
+     * Set label
+     *
+     * @param string $label
+     *
+     * @return Setting
      */
-    protected $container;
+    public function setLabel($label)
+    {
+        $this->label = $label;
 
+        return $this;
+    }
 
     /**
-     * Get id
+     * Get label
      *
-     * @return integer 
+     * @return string
      */
-    public function getId()
+    public function getLabel()
     {
-        return $this->id;
+        return $this->label;
+    }
+
+    /**
+     * Set key
+     *
+     * @param string $key
+     *
+     * @return Setting
+     */
+    public function setKey($key)
+    {
+        $this->key = $key;
+
+        return $this;
+    }
+
+    /**
+     * Get key
+     *
+     * @return string
+     */
+    public function getKey()
+    {
+        return $this->key;
+    }
+
+    /**
+     * Set type
+     *
+     * @param string $type
+     *
+     * @return Setting
+     */
+    public function setType($type)
+    {
+        $this->type = $type;
+
+        return $this;
+    }
+
+    /**
+     * Get type
+     *
+     * @return string
+     */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * Set value
+     *
+     * @param string $value
+     *
+     * @return Setting
+     */
+    public function setValue($value)
+    {
+        $this->value = $value;
+
+        return $this;
+    }
+
+    /**
+     * Get value
+     *
+     * @return string
+     */
+    public function getValue()
+    {
+        return $this->value;
     }
 
     /**
-     * Set name
+     * Set file
+     *
+     * @param \Enhavo\Bundle\MediaBundle\Entity\File $file
      *
-     * @param string $name
      * @return Setting
      */
-    public function setName($name)
+    public function setFile(\Enhavo\Bundle\MediaBundle\Entity\File $file = null)
```

Comment:
Please use Interface``Enhavo\Bundle\MediaBundle\Model\FileInterface`` instead of ``Enhavo\Bundle\MediaBundle\Entity\File`` in all methods